avm/ptn/app/container-job-toolkit/main.bicep

```diff
@@ -0,0 +1,341 @@
+          ]
+        : [],
+      // add passed secrets to the job. It uses the Managed Identity to access the secrets.
+      secrets ?? []
+    )
+    triggerType: 'Schedule'
+    scheduleTriggerConfig: {
+      cronExpression: cronExpression
+    }
+    registries: [
+      {
+        identity: services.outputs.userManagedIdentityResourceId
+        server: services.outputs.registryLoginServer
+      }
+    ]
+    containers: [
+      {
+        name: '${name}-scheduled-job'
+        image: import_image.outputs.importedImage.acrHostedImage
+        env: environmentVariables
+        resources: {
+          cpu: cpu
+          memory: memory
+        }
+      }
+    ]
+  }
+}
+
+// ============ //
+// Outputs      //
+// ============ //
+
+@description('The resource ID of the container job.')
+output resourceId string = job.outputs.resourceId
+
+@description('The name of the container job.')
+output name string = job.name
+
+@description('The name of the Resource Group the resource was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The resouce ID of the Log Analytics Workspace (passed as parameter value or from the newly created Log Analytics Workspace).')
+output logAnalyticsResourceId string = services.outputs.logAnalyticsResourceId
+
+@description('The principal ID of the user assigned managed identity.')
+output systemAssignedMIPrincipalId string = services.outputs.userManagedIdentityPrincipalId
+
+@description('Conditional. The virtual network resourceId, if a virtual network was deployed. If `deployInVnet` is `false`, this output will be empty.')
+output vnetResourceId string = services.outputs.vnetResourceId
+
+@description('Conditional. The address prefix for the private endpoint subnet, if a virtual network was deployed. If `deployInVnet` is `false`, this output will be empty.')
+output privateEndpointSubnetAddressPrefix string = deployInVnet
+  ? services.outputs.privateEndpointSubnetAddressPrefix
+  : ''
+
+@description('Conditional. The address prefix for the service endpoint subnet, if a virtual network was deployed. If `deployInVnet` is `false`, this output will be empty.')
+output deploymentscriptSubnetAddressPrefix string = services.outputs.deploymentscriptSubnetAddressPrefix
+
+@description('Conditional. The address prefix for the workload subnet, if a virtual network was deployed. If `addressPrefix` is empty, this output will be empty.')
+output workloadSubnetAddressPrefix string = services.outputs.workloadSubnetAddressPrefix
+
+// ================ //
+// Definitions      //
+// ================ //
+
+type environmentVariablesType = {
+  @description('Required. The environment variable name.')
+  name: string
+
+  @description('Conditional. The name of the Container App secret from which to pull the envrionment variable value. Required if `value` is null.')
+  secretRef: string?
+
+  @description('Conditional. The environment variable value. Required if `secretRef` is null.')
```

Comment:
What happens if both value and secretRef are provided? If either takes precedence, I'd suggest to specify that in the description

Reply:
That is unclear, as https://learn.microsoft.com/en-us/azure/container-apps/containers#configuration doesn't specify the order.

Reply:
That is unfortunately unclear, as not specified in the ContainerJob documentation.

Reply:
Fair enough, just wanted to mention so that you can give it a thought for next PRs. Resolving for this first version